internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -109,6 +109,10 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		desired[clusterv1.OwnerKindAnnotation] = owner.Kind
 		desired[clusterv1.OwnerNameAnnotation] = owner.Name
 	}
+	if machine.Spec.FailureDomain != nil {
```

Comment:
Could you go into some detail about why this is required? What's the use case for adding this annotation?

Reply:
This is a custom annotation on the node object is my understanding, so what component is consuming it to make scheduling decisions? What external component knows how to consume this annotation from the node?

Reply:
If I'm understanding right this PR should cover your use case: https://github.com/kubernetes-sigs/cluster-api/pull/7296 

It allows labels with specified prefixes to be reconciled from Machines to Node. It would be great if you could review the design proposal to ensure it would meet your requirements.

Reply:
@wongni What about the `topology.kubernetes.io/region` and `topology.kubernetes.io/zone` labels standardize by Kubernetes and aded by cloud-providers? Won't they work in your case?